### PR TITLE
Initial Commit 

### PR DESCRIPTION
**Pull request by cs-sameer-pathan**
PMDB Link: [click here](https://pmdb.fortinet.com/ProjectManagement/editprojects.php?id=24377)

#### Description:
New connector has been developed.

#### UTCs:
- [x] Verified connector installation.
- [x]  Verified connector logo.
- [x]  Verified checkhealth.
- [x]  Verified actions and playbooks list.
- [x]  Verified playbooks tags.

### CODE REVIEW

File: azure-key-vault/const.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
+""" Copyright start
```
Comment (cs-shubham-morankar) [inline azure-key-vault/const.py]: Update the name of the const.py to constants.py

File: azure-key-vault/microsoft_api_auth.py [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+  All rights reserved.
+  FORTINET CONFIDENTIAL & FORTINET PROPRIETARY SOURCE CODE
+  Copyright end """
+
+from requests import request
+from time import time, ctime
+from datetime import datetime
+from connectors.core.connector import get_logger, ConnectorError
+from connectors.core.utils import update_connnector_config
+from .const import *
```
Comment (cs-shubham-morankar) [inline azure-key-vault/microsoft_api_auth.py]: After changing the file name, please the "from .const import *" to "from .constants import *"

File: azure-key-vault/info.json [hunk 3]
```diff
@@ -0,0 +1,1038 @@
...
+          "name": "skip_token",
+          "type": "text",
+          "description": "(Optional) Skip Token is only used if a previous operation returned a partial result. If a previous response contains a nextLink element, the value of the nextLink element will include a skiptoken parameter that specifies a starting point to use for subsequent calls.",
+          "tooltip": "Skip Token is only used if a previous operation returned a partial result. If a previous response contains a nextLink element, the value of the nextLink element will include a skiptoken parameter that specifies a starting point to use for subsequent calls.",
+          "required": false,
+          "editable": true,
+          "visible": true
+        },
+        {
+          "title": "Size",
```
Comment (cs-shubham-morankar) [inline azure-key-vault/info.json:102]: Is there any default value? If Yes, please keep as default value.

**Review by cs-shubham-morankar — CHANGES_REQUESTED** [review]
Added review comments

File: azure-key-vault/info.json (hunk 3 above)
Comment (cs-sameer-pathan) [reply azure-key-vault/info.json:102]: There is no default value

File: azure-key-vault/operations.py [hunk 4]
```diff
@@ -0,0 +1,352 @@
...
+            logger.error('There was an error while handling the request')
+            raise ConnectorError('There was an error while handling the request')
+        except Exception as e:
+            logger.error('{0}'.format(e))
+            raise ConnectorError('{0}'.format(e))
+
+    def make_rest_call_vault(self, endpoint, params={}, data=None, method='POST'):
+        headers = {'Authorization': self.vault_token, 'Content-Type': 'application/json'}
+        service_url = self.server_url + endpoint
+        logger.debug('Request URL {}'.format(service_url))
```
Comment (cs-prerna-joshi) [inline azure-key-vault/operations.py]:
```suggestion
        logger.debug('Request URL {0}'.format(service_url))
```

File: azure-key-vault/operations.py [hunk 5]
```diff
@@ -0,0 +1,352 @@
...
+            raise ConnectorError('A connection error occurred')
+        except requests.exceptions.Timeout:
+            logger.error('The request timed out')
+            raise ConnectorError('The request timed out')
+        except requests.exceptions.RequestException:
+            logger.error('There was an error while handling the request')
+            raise ConnectorError('There was an error while handling the request')
+        except Exception as e:
+            logger.error('{0}'.format(e))
+            raise ConnectorError('{0}'.format(e))
```
Comment (cs-prerna-joshi) [inline azure-key-vault/operations.py]: The code is almost identical to make_rest_call. can we merge the both functions.

File: azure-key-vault/operations.py [hunk 6]
```diff
@@ -0,0 +1,352 @@
...
+
+def delete_key_vault(config, params):
+    kv = AzureKeyVault(config)
+    subscription_id = kv.subscription_id
+    resource_group = params.get('resource_group_name')
+    vault_name = params.get('vault_name')
+    endpoint = '/resourceGroups/{1}/providers/Microsoft.KeyVault/vaults/{2}'
+    endpoint = endpoint.format(subscription_id, resource_group, vault_name)
+    response = kv.make_rest_call(endpoint=endpoint, method='DELETE')
+    return {"status": response.status_code}
```
Comment (cs-prerna-joshi) [inline azure-key-vault/operations.py]:
```suggestion
    return {"status": response.status_code,
                  "result": "Successfully deleted the vault "}
```

File: azure-key-vault/operations.py [hunk 7]
```diff
@@ -0,0 +1,352 @@
...
+
+
+def delete_certificate(config, params):
+    config['scope'] = VAULT_SCOPE
+    kv = AzureKeyVault(config)
+    vault_name = params.get('vault_name')
+    payload = {}
+    certificate_name = params.get('certificate_name', '')
+    endpoint = '{0}.vault.azure.net/certificates/{1}'.format(vault_name, certificate_name)
+    response = kv.make_rest_call_vault(endpoint=endpoint, method='DELETE', params=payload)
```
Comment (cs-prerna-joshi) [inline azure-key-vault/operations.py]: why we are passing empty payload?

Comment (cs-sameer-pathan) [reply azure-key-vault/operations.py]: removed

File: azure-key-vault/operations.py (hunk 5 above)
Comment (cs-sameer-pathan) [reply azure-key-vault/operations.py]: yes, merged

File: azure-key-vault/info.json (hunk 3 above)
Comment (cs-sameer-pathan) [reply azure-key-vault/info.json:102]: No default value mentioned in documentation.

**Review by cs-prerna-joshi — APPROVED** [review]

**Pull request merged**